packages/angular/projects/cds-angular/src/cds/_stubs/module.ts.mustache

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+{{#directives}}
+import { {{directiveClassName}} } from './{{directiveFileName}}.directive';
+{{/directives}}
+
+import '@clr/core/{{moduleDirectory}}/register.js';
+
+@NgModule({
+  imports: [CommonModule],
+  declarations: [
+    {{#directives}}
+    {{directiveClassName}},
+    {{/directives}}
+  ],
+  exports: [
+    {{#directives}}
+    {{directiveClassName}},
+    {{/directives}}
+  ],
+})
+export class {{moduleClassName}} {}
```

Comment:
It looks like in the code this will generate one NgModule per component correct? I want to double check since Angular can only tree shake down to the NgModule level for us to keep the per component tree shaking apps should import each component NgModule instead of the root when possible.

```typescript
// app module
@NgModule({
  imports: [
    CommonModule,
    CdsIconModule,
    CdsAlertModule,
    CdsModalModule
  ],
```

Reply:
Yes, each core component will have its own module. And then there will be a single root module `CdsModule` that includes all of those component modules.